x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/components/legend.tsx

```diff
@@ -4,84 +4,145 @@
-            />
-          </EuiText>
-        </EuiFlexItem>
-      </EuiFlexGroup>
-    </EuiFlexItem>
-    <EuiFlexItem grow={false}>
-      <EuiFlexGroup gutterSize="xs" alignItems="center">
-        <EuiFlexItem grow={false}>
-          <span className="mlJobMapLegend__sourceNode" />
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiText size="xs" color="subdued">
-            <FormattedMessage
-              id="xpack.ml.dataframe.analyticsMap.legend.rootNodeLabel"
-              defaultMessage="source node"
-            />
-          </EuiText>
-        </EuiFlexItem>
-      </EuiFlexGroup>
-    </EuiFlexItem>
-  </EuiFlexGroup>
+const getJobTypeList = () => (
+  <>
+    <EuiListGroup>
+      <EuiListGroupItem iconType="outlierDetectionJob" label="Outlier detection" size="xs" />
+
+      <EuiListGroupItem iconType="regressionJob" label="Regression" size="xs" />
+
+      <EuiListGroupItem iconType="classificationJob" label="Classification" size="xs" />
+    </EuiListGroup>
+  </>
 );
+
+export const JobMapLegend: FC<{ theme: EuiThemeType }> = ({ theme }) => {
+  const [showJobTypes, setShowJobTypes] = useState<boolean>(false);
+
+  return (
+    <EuiFlexGroup className="mlJobMapLegend__container" alignItems="center">
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup gutterSize="xs" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <span className="mlJobMapLegend__sourceNode" />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiText size="xs" color="subdued">
+              <FormattedMessage
+                id="xpack.ml.dataframe.analyticsMap.legend.rootNodeLabel"
+                defaultMessage="source node"
+              />
+            </EuiText>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup gutterSize="xs" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <span className="mlJobMapLegend__indexPattern" />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiText size="xs" color="subdued">
+              <FormattedMessage
+                id="xpack.ml.dataframe.analyticsMap.legend.indexLabel"
+                defaultMessage="index"
+              />
+            </EuiText>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup gutterSize="xs" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <span className="mlJobMapLegend__transform" />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiText size="xs" color="subdued">
+              {JOB_MAP_NODE_TYPES.TRANSFORM}
+            </EuiText>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup gutterSize="xs" alignItems="center">
+          <EuiFlexItem grow={false}>
```

Comment:
Is there a reason why you moved the styling for this one out from the scss file? Is it possible to give the triangle in the legend the white background to match the map node and the other items in the legend?

![image](https://user-images.githubusercontent.com/7405507/102217333-58e00f80-3ed4-11eb-8615-0466d3e2e25f.png)

Reply:
It was moved to have access to the theme so the background color matches the background in dark/light mode.